core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerImpl.java

```diff
@@ -74,11 +87,12 @@ private AndroidMediaPlayerImpl(final AndroidMediaPlayerFacade mediaPlayer,
         forwarder.bind(listenersHolder.getInfoListeners());
 
         mediaPlayer.setOnPreparedListener(forwarder.onPreparedListener());
-        listenersHolder.addHeartbeatCallback(new CheckBufferHeartbeatCallback(forwarder.onHeartbeatListener()));
         mediaPlayer.setOnCompletionListener(forwarder.onCompletionListener());
         mediaPlayer.setOnErrorListener(forwarder.onErrorListener());
         mediaPlayer.setOnSizeChangedListener(forwarder.onSizeChangedListener());
 
+        bufferHeartbeatCallback.bind(forwarder.onHeartbeatListener());
```

Comment:
Inverted the dependency so that we could pass in the `BufferHeartbeatCallback` to `mock` and `verify` against.